src/websockets/TimerService/index.js

```diff
@@ -1,328 +1,293 @@
-          ...timerObject,
-          isRunning: false,
-          isUserPaused,
-          isApplicationPaused,
-          seconds: totalTimeAddedUp,
-        };
+// Here we set the goal and time to the goal time.
+const setGoal = (client, msg) => {
+  const goal = getGoal(msg);
+  client.goal = client.time = goal;
+};
 
-        /**
-         * Mongo Saving
-         *
-         * We would like to clean up here
-         * when we are saving persistent timer
-         * data. We only save when the user is
-         * logging out or all active connections have
-         * severed.
-         */
-        if (saveDataToDatabase) {
-          await setTimerToDatabase({
-            userId,
-            timerObject: {
-              // We use previous data since this is the correct time here
-              totalSeconds: totalTimeAddedUp,
-              ...newTimerObject,
-            },
-          });
+const goalOver10Hours = (client, time) => {
+  const goal = moment.duration(client.goal).add(time, 'milliseconds').asHours();
+  return goal > 10;
+};
 
-          await deleteTimerFromRedis({
-            userId,
-            redisClient: main,
-          });
-        } else {
-          // Resync timer to redis
-          await saveTimerToRedis({
-            userId,
-            redisClient: main,
-            timerObject: newTimerObject,
-          });
+/*
+Here we add the goal time.
+First we get the goal time from the message.
+Then we add it to the current goal time and set it as the new goal time.
+We also add it to the current time and set it as the new time.
+*/
+const addGoal = (client, msg) => {
+  const goal = getGoal(msg);
+  if (goalOver10Hours(client, goal)) return;
+
+  client.goal = moment
+    .duration(client.goal)
+    .add(goal, 'milliseconds')
+    .asMilliseconds()
+    .toFixed();
+  client.time = moment
+    .duration(client.time)
+    .add(goal, 'milliseconds')
+    .asMilliseconds()
+    .toFixed();
+};
 
-          // publish to all listening clients
-          await publish({
-            userId,
-            redisClient: publisher,
-            timerObject: newTimerObject,
-          });
-        }
+/*
+ * Here we check if the goal time is less than 15 minutes.
+ * */
+const goalLessThan15min = (client, time) => {
+  const goal = moment
+    .duration(client.goal)
+    .subtract(time, 'milliseconds')
+    .asMinutes();
+  return goal < 15;
+};
 
-        return;
-      } catch (e) {
-        logger.logException(e);
-      }
-    };
+/*
+ * Here we try to remove a goal time.
+ * First we get the goal time from the message.
+ * Then we subtract it from the current goal time and set it as the new goal time.
+ * We also subtract it from the current time and set it as the new time.
+ * If the new goal time is less than 15 minutes, we don't do anything.
+ * If the new time is less than 0, we set it to 0.
+ * */
+const removeGoal = (client, msg) => {
+  const goal = getGoal(msg);
+  if (goalLessThan15min(client, goal)) return;
+
```

Comment:
```suggestion

  if (!client.paused) {
    client.time = getTotalElapsedTime(client).asMilliseconds().toFixed();
    client.lastAccess = moment();
  }
  
```
Adding or removing time from the goal while the timer is running causes the timer to show an incorrect time for a second because client.time is only updated on pauses. This appears to fix it.

Reply:
Very very very good catch! Fixed!